File: binary_search/BinarySearch.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class BinarySearch {

    public static void main(String[] args) {
        ArrayList<Integer> integers = generateRandomList(1000);
        Collections.sort(integers);
        System.out.println(binarySearch(integers, 200));
        System.out.println(binarySearch(integers, 10000));
        System.out.println(binarySearch(integers, 100));

    }

    private static boolean binarySearch(final List<Integer> list, int value) {
        int mid = list.size() / 2;
        if(value == list.get(mid))
            return true;
        if (list.size() == 1)
            return value == list.get(0);
        if (value < list.get(mid))
            return binarySearch(list.subList(0, mid), value);
        else if (value > list.get(mid))
            return binarySearch(list.subList(mid, list.size()), value);
        return false;
    }

    private static ArrayList<Integer> generateRandomList(int size) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        Random random = new Random();
        for (int i = 0; i < size; i++)
            list.add(random.nextInt(1000));
        return list;
    }
}

File: binary_search/binary_search.py
import random


def binary_search(input_list, value):
    if input_list is None:
        return False
    mid = len(input_list) // 2
    if len(input_list) == 1:
        return value == input_list[0]
    if input_list[mid] == value:
        return True
    elif value < input_list[mid]:
        return binary_search(input_list[:mid], value)
    elif value > input_list[mid]:
        return binary_search(input_list[mid:], value)
    return False


if __name__ == '__main__':
    input_list = sorted(random.sample(range(1000), 500))
    print(binary_search(input_list, 100))
    print(binary_search(input_list, 10000))
    print(binary_search(input_list, 500))
